prefect_dask/utils.py

```diff
@@ -0,0 +1,167 @@
+"""
+Utils to use alongside prefect-dask.
+"""
+
+from contextlib import asynccontextmanager, contextmanager
+from datetime import timedelta
+from typing import Any, Dict, Optional, Union
+
+from distributed import Client, get_client
+from prefect.context import FlowRunContext, TaskRunContext
+
+from prefect_dask.exceptions import ImproperClientError
+
+
+def _populate_client_kwargs(
+    async_client: bool,
+    timeout: Optional[Union[int, float, str, timedelta]] = None,
+    **client_kwargs: Dict[str, Any],
+) -> Dict[str, Any]:
+    """
+    Helper method to populate keyword arguments for `distributed.Client`.
+    """
+    flow_run_context = FlowRunContext.get()
+    task_run_context = TaskRunContext.get()
+
+    if flow_run_context:
```

Comment:
Technically you should check the task run context first as both can exist in some situations.